src/parser/jobs/nin/modules/KunaisBaneWindow.tsx

```diff
@@ -130,16 +135,30 @@ export class TrickAttackWindow extends BuffWindow {
 		return window.data.find(cast => cast.action.id === this.data.actions.TEN_CHI_JIN.id) ? 1 : 0
 	}
 
-	private adjustExpectedRaitonCount(window: HistoryEntry<EvaluatedAction[]>, action: TrackedAction) {
-		const bufferedWindowStart = window.start - FIRST_TRICK_BUFFER
+	private adjustExpectedActionCount(window: HistoryEntry<EvaluatedAction[]>, action: TrackedAction) {
+		const bufferedWindowStart = window.start - FIRST_KUNAIS_BANE_BUFFER
 
 		if (action.action.id === this.data.actions.RAITON.id) {
-			// We push one Raiton outside the Trick window in the opener to fit an extra Bhava
+			// We push one Raiton outside the Kunai's Bane window in the opener to fit an extra Bhava
 			if (bufferedWindowStart <= this.parser.pull.timestamp) {
 				return -1
 			}
 		}
 
+		if (action.action.id === this.data.actions.TENRI_JINDO.id) {
+			// Tenri Jindo should only be expected when the window contains TCJ
+			if (window.data.find(value => value.action.id === this.data.actions.TEN_CHI_JIN.id) !== undefined) {
```

Comment:
I don't think this check is behaving correctly, this log has 1/0 expected on the TCJ windows, and 0/1 on the non-TCJ windows, instead of 1/1 and 0/0:
http://127.0.0.1:3000/fflogs/Mbd9P6V4fvCN8tDG/4/293

Reply:
I feex.